src/ascii/interger.cairo

```diff
@@ -0,0 +1,220 @@
+use core::option::OptionTrait;
+use alexandria::data_structures::array_ext::ArrayTraitExt;
+use array::{ArrayTrait, Array};
+use traits::{Into, TryInto, DivRem};
+use zeroable::Zeroable;
+
+trait IntergerToAsciiTrait<T, U> {
+    fn to_ascii(self: T) -> U;
+}
+
+// converts intergers into an array of its individual ascii values
+trait IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252>;
+    fn to_inverse_ascii_array(self: T) -> Array<felt252>;
+}
+
+// converts intergers into an array of its individual ascii values
+// e.g. 123 -> [49, 50, 51]
+impl IntergerToAsciiArrayTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiArrayTrait<T> {
+    fn to_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = self.to_inverse_ascii_array();
+        new_arr.reverse();
+        new_arr
+    }
+
+    fn to_inverse_ascii_array(self: T) -> Array<felt252> {
+        let mut new_arr = ArrayTrait::new();
+        if self <= 9.try_into().unwrap() {
+            new_arr.append(self.into() + 48);
+            return new_arr;
+        }
+
+        let mut num = self;
+        loop {
+            if num.is_zero() {
+                break ();
+            }
+            let (quotient, remainder) = DivRem::div_rem(
+                num, TryInto::<felt252, T>::try_into(10).unwrap().try_into().expect('Division by 0')
+            );
+            new_arr.append(remainder.into() + 48);
+            num = quotient;
+        };
+        new_arr
+    }
+}
+
+// gneric implementation for small intergers <u128 
+// to transform its intergers into a string represented as a single felt252
+// e.g. 1000 -> "1000"
+impl SmallIntergerToAsciiTraitImpl<
+    T,
+    impl TNumericLiteral: NumericLiteral<T>,
+    impl TPartialOrd: PartialOrd<T>,
+    impl TDivRem: DivRem<T>,
+    impl TInto: Into<T, felt252>,
+    impl TTryInto: TryInto<felt252, T>,
+    impl TTryIntoZero: TryInto<T, NonZero<T>>,
+    impl TZeroable: Zeroable<T>,
+    impl TDrop: Drop<T>,
+    impl TCopy: Copy<T>,
+> of IntergerToAsciiTrait<T, felt252> {
+    fn to_ascii(self: T) -> felt252 {
+        if self <= 9.try_into().unwrap() {
+            return self.into() + 48;
+        }
+
+        let inverse_ascii_arr = self.to_inverse_ascii_array();
+        let len = inverse_ascii_arr.len();
+        let mut index = 0;
+        let mut ascii: felt252 = 0;
+        loop {
+            if index >= len {
+                break ();
+            }
+            // recursively keep getting the index from the end of the array
+            let l_index = len - index - 1;
+            ascii = ascii * 256 + *inverse_ascii_arr[l_index];
+            index += 1;
+        };
```

Comment:
You'll need to update imports `use array::{ArrayTrait, SpanTrait};`

This part could be updated to:
```suggestion
        let mut inverse_ascii_arr = (self.to_inverse_ascii_array()).span();
        let mut ascii: felt252 = 0;
        loop {
          match inverse_ascii_arr.pop_back() {
              Option::Some(val) => {
                  ascii = ascii * 256 + *val;
              },
              Option::None(_) => {
                  break;
              },
          };
```